tortoise/tortoise.go

```diff
@@ -291,6 +285,19 @@ func (t *turtle) firstDisagreement(ctx context.Context, blid types.LayerID, ball
 	start = t.ballotLayer[base]
 
 	for lid := start; lid.Before(blid); lid = lid.Add(1) {
+		if len(t.blocks[lid]) == 0 {
+			// it is not possible to encode against for empty layer
+			// because there are no blocks
+			// therefore we should not pick base ballot that votes abstain on
+			// layer outside zdist if layer is empty
+			if !isUndecided(t.Config, t.decided, lid, last) && t.full.abstained(ballotID, lid) {
+				t.logger.With().Debug("ballot votes abstain on a layer without blocks. can't use as a base ballot",
+					ballotID,
+					lid,
+				)
+				return types.LayerID{}, nil
```

Comment:
returning layer id 0, instead of lid, here effectively move this ballot to the back of the base ballot sort?

Reply:
yes